src/screens/reader/components/ChapterDrawer.js

```diff
@@ -0,0 +1,229 @@
+import React, { useRef, useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { Text, TouchableRipple } from 'react-native-paper';
+import color from 'color';
+import { useTheme } from '@hooks/useTheme';
+import { FlashList } from '@shopify/flash-list';
+import { Button } from '@components/index';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { LinearGradient } from 'expo-linear-gradient';
+import { getString } from '@strings/translations';
+import { useNovel } from '@hooks/reduxHooks';
+
+const ChapterDrawer = ({ state, navigation }) => {
+  const theme = useTheme();
+  const insets = useSafeAreaInsets();
+  const styles = createStylesheet(theme, insets);
+
+  let listRef = useRef();
+  const { chapters } = useNovel();
```

Comment:
Chapters is set when the User open the Novel Details screen, what's the behaviour when they navigate from other screens?

Reply:
Problem is now resolved

Now, if you navigate from other screens it dispatches setNovel and getNovelAction. It works that way, however I don't think it is the best solution, but the best solution I could come up with.

Also it now adopts the list preferences from the novelscreen. (sort and filter)